ci/tdx_canonical_linux_stack.sh

```diff
@@ -25,168 +33,24 @@ if [[ $DISTRO_VER == "23.10" ]]; then
-        echo "creating TD guest with libvirt"
-        grep -q 'user =' "$LIBVIRT_CONF" && sed 's/^user =.*/user = \"root\"/' -i "$LIBVIRT_CONF" || sh -c "echo 'user = "root"' >> $LIBVIRT_CONF"
-        grep -q 'group =' "$LIBVIRT_CONF" && sed 's/^group =.*/group = \"root\"/' -i "$LIBVIRT_CONF" || sh -c "echo 'group = "root"' >> $LIBVIRT_CONF"
-        grep -q 'dynamic_ownership =' "$LIBVIRT_CONF" && sed 's/^dynamic_ownership =.*/dynamic_ownership = 0/' -i "$LIBVIRT_CONF" || sh -c "echo 'dynamic_ownership = 0' >> $LIBVIRT_CONF"
-        systemctl restart libvirtd
-        cd "$GUEST_TOOLS_DIR"
-        cleanup
-        var=$(./${TD_VIRSH_BOOT_CMD})
-        ret=$?
-        echo $var
-        if [ $ret -ne 0 ]; then
-                exit 1
-        fi
-        sleep 20
-        echo "verifying TD guest on libvirt"
-        if [[ $DISTRO_VER == "23.10" ]]; then
-                port_num=$(echo $var | awk -F '-p' '{print $2}' | cut -d ' ' -f 2)
-        else
-                port_num=$(echo $(./tdvirsh list --all) | awk -F 'ssh:' '{print $2}' | cut -d ',' -f 1)
-                vm_name=$(echo $var | awk -F 'Name:' '{print $2}' | cut -d ' ' -f 2)
-        fi
-        verifytd $port_num
-        del_vm=$(./tdvirsh delete ${vm_name})
-        echo $del_vm
-}
-
-setup_pycloudstack(){
-        echo "Installing PyCloudStack required packages in Guest image"
-        LIBGUESTFS_DEBUG=1 LIBGUESTFS_TRACE=1 virt-customize -a $QCOW2_IMG --install qemu-utils,libguestfs-tools,cpuid,python3-virtualenv,python3-libvirt,libguestfs-dev,libvirt-dev,python3-dev,net-tools,qemu-guest-agent,docker.io,cgroupfs-mount --run-command 'cgroupfs-mount' --run-command 'dockerd -D &' --run-command "sed -i 's/169.254.2.3/127.0.0.53/g' /etc/resolv.conf"
-        echo "Installing PyCloudStack required packages on host system"
-        apt install -y python3-virtualenv python3-libvirt libguestfs-dev libvirt-dev python3-dev net-tools
-        usermod -aG libvirt root
-        systemctl restart libvirtd
-        echo "Setting up PyCloudStack venv"
-        [ -d $TDX_TOOLS_DIR ] && rm -rf $TDX_TOOLS_DIR
-        git clone https://github.com/anjalirai-intel/tdx-tools.git $TDX_TOOLS_DIR
-        cd $TDX_TOOLS_DIR/tests/
-        source setupenv.sh
-        ssh-keygen -f tests/vm_ssh_test_key -N ""
-        cat >artifacts.yaml <<EOL
-  latest-guest-image-ubuntu:
-   source: file://${QCOW2_IMG}
-  latest-guest-kernel-ubuntu:
-   source: file://${VMLINUZ}
-EOL
-
-        cat artifacts.yaml
-}
-
-run_pycloudstack(){
-        TEST_TYPE=$1
-        cd $TDX_TOOLS_DIR/tests/
-        if [[ $TEST_TYPE == "sanity" ]]; then
-                ./run.sh -g ubuntu -c tests/test_tdvm_lifecycle.py
-        else
-                ./run.sh -g ubuntu -s all
-        fi
-        if [ $? -ne 0 ]; then
-                exit 1
-        fi
-}
-
-setup_canonical_suite(){
-	echo "Installing Canonical suite required packages on host system"
-	sudo apt install tox python3
-}
+if [[ -z "$JENKINS_URL" ]]; then
```

Comment:
What's the intention behind this check?

Reply:
To ensure that shell script is invoke via jenkins pipeline